### PR TITLE
Metadata Support

### PR DESCRIPTION
**Pull request by bebon901**
metadata files can be viewed in /var/log/rpi-sb-provisioner/<serial>/metadata/<serial>.json
Added basic parsing of metadata file.

NOTE: Needs updated version of rpiboot - specify this in nfpm.yaml?

### CODE REVIEW

File: key-writer/keywriter.sh [hunk 1]
```diff
@@ -2,10 +2,9 @@
 
 set -x
 
+. /etc/rpi-sb-provisioner/config
```
Comment (tdewey-rpi) [inline key-writer/keywriter.sh:5]: Undo this, please.

File: key-writer/keywriter.sh [hunk 2]
```diff
@@ -209,8 +209,62 @@ echo "eeprom_write_protect=1" >> "${FLASHING_DIR}/config.txt"
 fi
 
 # With the EEPROMs configured and signed, RPIBoot them.
-[ -z "${DEMO_MODE_ONLY}" ] && rpiboot -d "${FLASHING_DIR}" -i "${TARGET_DEVICE_SERIAL}"
+mkdir -p "/var/log/rpi-sb-provisioner/${TARGET_DEVICE_SERIAL}/metadata/"
+[ -z "${DEMO_MODE_ONLY}" ] && rpiboot -d "${FLASHING_DIR}" -i "${TARGET_DEVICE_SERIAL}" -j "/var/log/rpi-sb-provisioner/${TARGET_DEVICE_SERIAL}/metadata/"
 
 touch "${DEVICE_SERIAL_STORE}/${TARGET_DEVICE_SERIAL}"
 
+USER_BOARDREV="0x$(cat /var/log/rpi-sb-provisioner/94a97b1d/metadata/94a97b1d.json | jq -r '.USER_BOARDREV')"
```
Comment (tdewey-rpi) [inline key-writer/keywriter.sh:217]: Replace with variable

**Comment by roliver-rpi** [comment]
Are these commits missing some changes? One of the commit messages has "Added hooks in provisioner.sh to help the app" but provisioner.sh is not modified by this MR.

**Comment by tdewey-rpi** [comment]
> Are these commits missing some changes? One of the commit messages has "Added hooks in provisioner.sh to help the app" but provisioner.sh is not modified by this MR.

Which commit?

**Comment by roliver-rpi** [comment]
> > Are these commits missing some changes? One of the commit messages has "Added hooks in provisioner.sh to help the app" but provisioner.sh is not modified by this MR.
> 
> Which commit?

5835727d8f163736ce3f1a27af7ebc9da9fe567e claims "Added hooks in provisioner.sh to help the app". Some of the other commit messages could do with reviewing as well as they don't align well with the contents of the commit.

**Comment by tdewey-rpi** [comment]
ACK. @bebon901 please review the commit messages in this series, or are we good to squash all these commits into one as part of a merge (with an appropriate message?)

At the very least - this now needs a rebase.

File: monitor/systemctl_python.py [hunk 3]
```diff
@@ -64,13 +68,20 @@ def list_failed_devices():
...
-                f = open("/var/log/rpi-sb-provisioner/" + device + "/finished", "r")
-                status = f.read()
-                if "1" in status:
-                    failed_devices.append(device)
-                f.close()
+        provisioner_success = -1
+        keywriter_success = -1
+        if path.exists("/var/log/rpi-sb-provisioner/" + device + "/progress"):
+            f = open("/var/log/rpi-sb-provisioner/" + device + "/progress", "r")
+            status = f.read()
```
Comment (tdewey-rpi) [inline monitor/systemctl_python.py:75]: Does this consume the entire file?

**Pull request merged**